core/src/banking_stage/transaction_scheduler/scheduler_metrics.rs

```diff
@@ -332,3 +334,71 @@ impl SchedulerTimingMetricsInner {
         self.receive_completed_time_us = 0;
     }
 }
+
+#[derive(Default)]
+pub struct SchedulerLeaderDetectionMetrics {
+    inner: Option<SchedulerLeaderDetectionMetricsInner>,
+}
+
+pub struct SchedulerLeaderDetectionMetricsInner {
+    slot: Slot,
+    bank_detected_time: Instant,
+    bank_detected_delay_us: u64,
+}
+
+impl SchedulerLeaderDetectionMetrics {
+    pub fn update_and_maybe_report(&mut self, bank_start: Option<&BankStart>) {
+        match (&self.inner, bank_start) {
+            (None, Some(bank_start)) => self.initialize_inner(bank_start),
+            (Some(_inner), None) => self.report_and_reset(),
+            (Some(inner), Some(bank_start)) if inner.slot != bank_start.working_bank.slot() => {
+                self.report_and_reset();
+                self.initialize_inner(bank_start);
+            }
+            _ => {}
```

Comment:
nit: i prefer `(None, None) | (Some(_), Some(_))` for quick human readability. i had to pause a bit what cases for which this match arms falls under.

Reply:
So if I do `(None, None) | (Some(_inner), Some(_bank_start))` compiler complains because the vars `_inner` and `_bank_start` are not declared in both cases.
If I do `(None, None) | (Some(_), Some(_))` it will compile, but I feel it is a bit confusing; we do not want `Some(_), Some(_))` in all cases - only when `inner.slot == bank_start.working_bank.slot()`

I think I will opt to keep what I had; I have 3 cases I care about and do nothing in all other cases.